src/main/java/patternverifiers/SingletonVerifier.java

```diff
@@ -1,17 +1,312 @@
+
+        // Finds and checks if the constructors are private or not.
+        compUnit.findAll(ConstructorDeclaration.class).forEach(constructor -> {
+            if (constructor.isPrivate()) {
+                listOfIsPrivate.add(true);
+            } else {
+                listOfIsPrivate.add(false);
+            }
+        });
+
+        // Takes all boolean values and ANDs them to one value.
+        for (Boolean bool : listOfIsPrivate) {
+            isPrivConstr &= bool;
+        }
+
+        return isPrivConstr;
+    }
+
+    /**
+     * Method for declaring if a java class holds a field variable of a static instance
+     * https://stackoverflow .com/questions/53300710/how-to-parse-inner-class-from-java-source-code
+     * might help solving a check for inner classes
+     *
+     * @param compUnit The CompilationUnit representing the java class to look at
+     *
+     * @return True iff the java class holds a field variable with a static modifier of the same
+     *     type as the class itself (eg. static SingletonVerifier sv;)
+     */
+    public boolean hasAStaticInstance(CompilationUnit compUnit) {
+        boolean isStatic = false;
+        boolean isPrivate = false;
+        boolean isInstantiated = false;
+
+        for (FieldDeclaration fieldDeclaration : VariableReader.readVariables(
+            compUnit)) {      // For each FieldDeclaration in the java file
+            if (fieldDeclaration.getVariables().get(0).getType().toString().equals(
+                compUnit.getType(0).getNameAsString())) {    // If there is a field of the
+                // same type as the file itself, probably needs to check for
+                // several different classes in the same file, can have inner
+                // classes etc not sure how javaparser handles that.
+                if (isStatic && isPrivate) {  // If an instance variable has already been found,
+                    // return false since a singleton should only have one instance.
+                    isStatic = false;
+                    isPrivate = false;
+                    return false;
+                }
+                if (!fieldDeclaration.getVariables().get(0).getInitializer().isEmpty()) {
+                    isInstantiated = true;
+                }
+                for (Modifier md : fieldDeclaration
+                    .getModifiers()) {     // For each modifier on that field
+                    if (md.getKeyword().asString().equals(
+                        "static")) {  // If that modifier is static
+                        isStatic = true;
+                    } else if (md.getKeyword().asString().equals(
+                        "private")) { // Else if that modifier is private
+                        isPrivate = true;
+                    }
+                }
+                instanceVar = fieldDeclaration;
+            }
+        }
+        return isStatic && isPrivate && isInstantiated; // should isInstansiated be here?
```

Comment:
isInstansiated was not placed here when me and nemo started merging develop into this branch. We found that it was unused and it seemed like it was supposed to be here. Is that correct? @Darkarca

Reply:
Oops oversight on my part, isInstantiated can be set to true in two different methods but I seem to have missed adding it in the other method, resolved in next commit!

Reply:
Clarification, isInstantiated should be checked for in verify, but it can be set to true in two different method. Hence I made the variable outside of the method and added a check in verify